contrib/api.sh

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env bash
+#
+# Script for querying the API.
+#
+# Shellcheck can't search dynamic paths
+# shellcheck source=/dev/null
+
+set -euo pipefail
+
+file=""                         # File name of the all-features API text file.
+crate_full_name=""              # Full crate name using underscores e.g., `bitcoin_primitives`.
+crate=""                        # Short name e.g., `primitives`.
+
+# Set to false to turn off verbose output.
+flag_verbose=false
+
+usage() {
+    cat <<EOF
+Usage:
+
+    ./api.sh CRATE COMMAND
+
+CRATE
+  - hashes          bitcoin_hashes
+  - io              bitcoin-io
+  - primitives      bitcoin-primitives
+  - units           bitcoin-units
+
+CMD
+  - types             Show all public types (structs and enums)
+  - types_no_err      Show all public types (structs and enums) excluding error types.
+EOF
+}
+
+main() {
+    local _crate="${1:-usage}"
+    local _cmd="${2:-usage}"
+
+    # FIXME: This is a hackish way to get the help flag.
+    if [ "$_crate" = "usage" ] || [ "$_cmd" = "usage" ] || [ "$_cmd" = "-h" ] || [ "$_cmd" = "--help" ]; then
```

Comment:
In 302a7018f1e4d62b407a6ad29dd24684aec6b27e:

I would suggest just always dumping the usage after "unsupported crate". Then you can special-case `-h` and `--help` up here if you want.

Reply:
Cheers, the `api` script is pretty messy still but it shell checks cleanly.